ekncontent/ekncontent/eknc-query-object.c

```diff
@@ -825,3 +826,68 @@ eknc_query_object_get_sort_value (EkncQueryObject *self)
       break;
     }
 }
+
+/**
+ * eknc_query_object_new_from_object:
+ * @source: the query object to copy
+ * @first_property_name: name of the first property to set
+ * @...: values and names of other properties to set
+ *
+ * Clone all properties not explicitly set from the source object.
+ *
+ * Returns: (transfer full): the newly allocated query object
+ */
+EkncQueryObject *
+eknc_query_object_new_from_object (EkncQueryObject *source,
+                                   const gchar     *first_property_name,
+                                   ...)
```

Comment:
Yup, its in the overrides in initial commit porting query object to C.

I could squash this commit into the initial port, but didn't want to bloat the initial query object commit too much cause there's so much there. The C version is not needed till we port xapian bridge over.